apps/dashr-wind-steaming/app.R

```diff
@@ -0,0 +1,443 @@
+      layout(
+        plot_bgcolor = app_color[["graph_bg"]],
+        paper_bgcolor = app_color[["graph_bg"]],
+        font = list("color" = "#fff"),
+        xaxis <- list(
+          range = list(0, 200),
+          showline = TRUE,
+          zeroline = FALSE,
+          fixedrange = TRUE,
+          tickvals = list(0, 50, 100, 150, 200),
+          ticktext = list("200", "150", "100", "50", "0"),
+          title = "Time Elapsed (sec)"
+        ),
+        yaxis <- list(
+          range = list(
+            min(0, min(df$Speed)),
+            max(45, max(df$Speed) + max(df$SpeedError))
+          ),
+          showgrid = TRUE,
+          showline = TRUE,
+          fixedrange = TRUE,
+          zeroline = FALSE,
+          gridcolor = app_color[["graph_line"]],
+          nticks = max(6, round(tail(df$Speed, n = 1)) / 10)
+        )
+      )
+    return(fig)
+  }
+)
+
+app$callback(
+  output = list(id = "wind-direction", property = "figure"),
+  params = list(input(id = "wind-speed-update", property = "n_intervals")),
+  
+  function(interval){
+    # Generate the wind direction graph.
+    # params interval: update the graph based on an interval
+    
+    total_time <- get_current_time()
+    df <- get_wind_data_by_id(total_time)
+    val <- tail(df$Speed, n=1)
+    direction <- list(0, (df$Direction[1] - 20), (df$Direction[1] + 20), 0)
+    
+    fig <- plot_ly(
+      type = "scatterpolar",
+      mode = "lines",
+      height = 350
+    ) %>%
+      add_trace(
+        r = list(0, val, val, 0),
+        theta = direction,
+        fillcolor = "#084E8A",
+        fill = "toself",
+        line = list(color = "rgba(32, 32, 32, .6)", "width" = 1)
+      ) %>%
+      add_trace(
+        r = list(0, val * 0.65, val * 0.65, 0),
+        theta = direction,
+        fillcolor = "#B4E1FA",
+        fill = "toself",
+        line = list(color = "rgba(32, 32, 32, .6)", "width" = 1)
+      ) %>%
+      add_trace(
+        r = list(0, val * 0.3, val * 0.3, 0),
+        theta = direction,
+        fillcolor = "#EBF5FA",
+        fill = "toself",
+        line = list(color = "rgba(32, 32, 32, .6)", "width" = 1)
+      ) %>% 
+      layout(
+        plot_bgcolor = app_color[["graph_bg"]],
+        paper_bgcolor=app_color[["graph_bg"]],
+        font = list("color" = "#fff"),
+        autosize = FALSE,
+        polar = list(
+          "bgcolor" = app_color[["graph_line"]],
+          "radialaxis" = list("range" = c(0, 45), "angle" = 45, "dtick" = 10),
+          "angularaxis" = list("showline" = FALSE, "tickcolor" = "white")
+        ),
+        showlegend = FALSE
+      )
+    return(fig)
+  }
+)
+
+app$callback(
+  output = list(id = "wind-histogram", property = "figure"),
+  params = list(
+    input(id = "wind-speed-update", property = "n_intervals"),
+    list(state(id = "wind-speed", property = "figure"),
+         state(id = "bin-slider", property = "value"),
+         state(id = "bin-auto", property = "values"))
+  ),
```

Comment:
This throws an error:

`Error: Callback parameters must be inputs or states. Please verify formatting of callback parameters.`

Does this work as you intended?
```
app$callback(
  output = list(id = "wind-histogram", property = "figure"),
  params = list(
    input(id = "wind-speed-update", property = "n_intervals"),
    state(id = "wind-speed", property = "figure"),
    state(id = "bin-slider", property = "value"),
    state(id = "bin-auto", property = "values")),
...
```